src/modules/fancyzones/FancyZonesLib/DraggingState.cpp

```diff
@@ -72,12 +74,23 @@ void DraggingState::OnMouseDown()
     m_keyUpdateCallback();
 }
 
+void DraggingState::OnMiddleMouseDown()
+{
+    if (!this->IsDragging())
+    {
+        return;
+    }
+
+    m_middleMouseState = !m_middleMouseState;
+    m_keyUpdateCallback();
+}
+
 bool DraggingState::IsDragging() const noexcept
 {
     return m_dragging;
 }
 
 bool DraggingState::IsSelectManyZonesState() const noexcept
 {
-    return m_ctrlKeyState.state();
+    return m_ctrlKeyState.state() ^ m_middleMouseState;
```

Comment:
Minor question: why not `or`?

Reply:
That's me being lazy and copying the code from `UpdateDraggingState`. I agree an `or` is more readable and clear.

Reply:
Resolved.